src/js/export.js

```diff
@@ -0,0 +1,212 @@
+
+    csvRow += taxonomy
+      .map(key => entryTaxonomy[key])
+      .map(val => val ? val.join(leafDelimiter) : "")
+      .join(CSVDelimiter);
+
+    csvRow += "\n";
+
+    return csvRow;
+  }
+
+  function exportToCSV(filename, csvContent) {
+      var blob = new Blob([csvContent], { type: 'text/csv;charset=utf-8;' });
+      if (navigator.msSaveBlob) { // IE 10+
+          navigator.msSaveBlob(blob, filename);
+      } else {
+          var link = document.createElement("a");
+          if (link.download !== undefined) { // feature detection
+              // Browsers that support HTML5 download attribute
+              var url = URL.createObjectURL(blob);
+              link.setAttribute("href", url);
+              link.setAttribute("download", filename);
+              link.style.visibility = 'hidden';
+              document.body.appendChild(link);
+              link.click();
+              document.body.removeChild(link);
+          }
+      }
+  }
+
+  function createExportModal(){
+    modal =
+    el('div.modal', [
+        el('div', [
+            el('div.close-btn#exportCloseBtn', ['']),
+            el("h2", [`Export collection ${cName} (#${cID})` ]),
+            el("div#exportFilenameWrapper", [
+                el("input.modal-input-box#exportFilename", {type:"text", placeholder:"Filename"}),
+                el("label", [" .csv"])
+            ]),
+
+            el("div.modal-divider"),
+            delimiterDiv("CSV", "Select CSV delimiter "),
+            el("div.modal-spacing"),
+            delimiterDiv("Leaf", "Select taxonomy leaf delimiter "),
+            el("div.modal-divider"),
+
+            el("div", [
+                el('button#exportBtn.btn', ["Export"]),
+                el('button#exportCancelBtn.btn', ["Cancel"]),
+            ])
+        ])
+    ]);
+    setTimeout(() => modal.classList.add("appear"), 100)
+    document.body.appendChild(modal);
+  }
+
+  function delimiterDiv(name, text){
+    return  el("div", {id: "delimiter" + name}, [
+                el("div.export-div-delimiter." + name, [
+                    el("label", [text]),
+                    el("select.export-select-delimiter", {id: "selectDelimiter" + name}, [
+                        delimiters.map(delimiter =>
+                        el('option', { value: delimiter.value }, [ delimiter.display ])),
+                    ])
+                ])
+            ])
+  }
+
+  function isFilenameValid(){
+    if(filename === ""){
+      document.getElementById("exportFilenameWrapper").appendChild(
+          el("div.complaint.export", {text:"Please supply information"})
+      );
+      return false;
+    }
+    return true;
+  }
+
+  function areDelimitersValid(){
+    CSVDelimiter = selectDelimiterCSV.value;
+    leafDelimiter = selectDelimiterLeaf.value;
+    if(CSVDelimiter === leafDelimiter){
+      document.getElementById("delimiterCSV").appendChild(
+          el("div.complaint.export", {text:"The delimiters have to be different"})
```

Comment:
`el("div.complaint.export", ["The delimiters have to be different"])` will append a text node to the dom object. Might be some difference in styling but I don't think so.

Anyway, since two functions append text you should probably add a helper function: `function complain(node, txt)`